draft-ietf-quic-qlog-main-schema.md

```diff
@@ -244,6 +246,57 @@ schema structured as:
     * `scenario` event type
     * `marker` event type
 
+# Abstract LogFile Class {#abstract-logfile}
+
+A Log file is intended to contain a collection of events that are in some way
+related. An abstract LogFile class containing fields common to all log files is
+defined. Each concrete log file format derives from this, extending it by
+defining semantics and any custom fields.
+
+~~~ cddl
+LogFile = {
+    file_schema: text
+    qlog_serialization_format: text
+    ? title: text
+    ? description: text
+    event_schema: [+text]
+
+    ; can contain any amount of custom fields
+    * text => any
+}
+~~~
+{: #abstract-logfile-def title="LogFile definition"}
+
+The required "file_schema" field identifies the concrete log file format. It
+MUST have a value that is an absolute URI; see {{schema-uri}} for rules and
+guidance. In order to make it easier to parse and identify qlog files and their
+serialization format, the "file_schema" and "qlog_serialization_format" fields
+and their values SHOULD be in the first 256 characters/bytes of the resulting
+log file.
+
+The required "qlog_serialization_format" field indicates the serialization
+format using a media type {{!RFC2046}}. It is case-insensitive.
+
+The optional "title" and "description" fields provide additional free-text
+information about the file.
+
+The required "event_schema" field is described in {{event-types-and-schema}}.
```

Comment:
Give a short description here as well about the goal of this field (just so readers get a high-level overview from this section)